### PR TITLE
[FIX] web: format local week year

### PR DESCRIPTION
**Pull request by lvsz**
Versions
--------
- saas-17.4+

Enterprise PR: https://github.com/odoo/enterprise/pull/73318

Steps
-----
1. Have Planning installed;
2. change localisation to 'en_GB' (or any whose weeks start on Monday);
3. open in Gantt view in weekly granularity;
4. go to the week of 2024-12-30.

Issue
-----
Label on top right displays "W1 2024";

Cause
-----
It uses the `getLocalWeekNumber` function to get the week number of 2024-12-30, which falls in the first week of 2025. It then combines it with the `year` of the date, which is 2024.

Solution
--------
### Community:
Like the `weeknumber` function added in 9c47e911d0ca7 to `odoo.tools.date_utils`, have a function in `web` that returns both year and week number.

### Enterprise:
Define a `formatLocalWeekYear` function in `web_gantt` using the new `getLocalYearAndWeek` function from `web`.

opw-4280192

### CODE REVIEW

**Comment by lvsz** [comment]
Closed for simpler fix in Enterprise.

**Comment by lvsz** [comment]
Simple `kkkk` approach was a dud, doesn't work for non-ISO locales like `en_US`.

File: addons/web/static/src/core/l10n/dates.js [hunk 1]
```diff
@@ -398,6 +407,20 @@ export function formatDuration(seconds, showFullDuration) {
...
+ *
+ * @param {NullableDateTime} value
+ * @returns {string}
+ */
+export function formatLocalWeekYear(value) {
+    if (!value) {
+        return "";
+    }
+    const { year, week } = getLocalYearAndWeek(value);
+    return `W${week} ${year}`;
```
Comment (lvsz) [inline addons/web/static/src/core/l10n/dates.js:421]:
```suggestion
    return `${_t("Week")[0]}${week} ${year}`;
```
We could localize this harder by taking the first letter of `_t("Week")`, but I'm only fluent in languages where "Week" starts with "W", so no idea whether "W52" already makes sense in e.g. Mandarin or French.

**Review by brboi — COMMENTED** [review]
Nicely done ;-)

Please apply the requested changes and also write a test on the gantt side.

Before validating the fw-port in master, it would be nice to mark as deprecated the `getLocalWeekNumber` function, or just ping me in order to remove it directly and adapt the calendar usages (unless you want to do it yourself ;-))

Thanks for investigating this and to have come up with a clean solution.

**Comment by lvsz** [comment]
> Please apply the requested changes and also write a test on the gantt side.

Alright, done.

> Before validating the fw-port in master, it would be nice to mark as deprecated the `getLocalWeekNumber` function, or just ping me in order to remove it directly and adapt the calendar usages (unless you want to do it yourself ;-))

Sure, will do one of those.

**Review by brboi — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**